tests/php/src/Validation/URLValidationProviderTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace AmpProject\AmpWP\Tests\Validation;
+
+use AmpProject\AmpWP\Tests\Helpers\AssertContainsCompatibility;
+use AmpProject\AmpWP\Tests\Helpers\PrivateAccess;
+use AmpProject\AmpWP\Tests\Helpers\ValidationRequestMocking;
+use AmpProject\AmpWP\Validation\URLValidationProvider;
+use WP_UnitTestCase;
+
+/** @coversDefaultClass URLValidationProvider */
+final class URLValidationProviderTest extends WP_UnitTestCase {
+	use PrivateAccess, AssertContainsCompatibility;
+
+	/**
+	 * Setup.
+	 *
+	 * @inheritdoc
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->validation_provider = new URLValidationProvider( 100 );
+		add_filter( 'pre_http_request', [ ValidationRequestMocking::class, 'get_validate_response' ] );
+	}
+
+	/**
+	 * Test get_url_validation.
+	 *
+	 * @covers ::get_url_validation()
+	 */
+	public function test_get_url_validation() {
+		$single_post_permalink = get_permalink( self::factory()->post->create() );
+		$this->call_private_method( $this->validation_provider, 'get_url_validation', [ $single_post_permalink, 'post' ] );
```

Comment:
This is not a private method.
```suggestion
		$this->validation_provider->get_url_validation( $single_post_permalink, 'post' );
```